Xamarin/SSW.Rewards/SSW.Rewards/ViewModels/LoginPageViewModel.cs

```diff
@@ -141,7 +141,14 @@ private async void Refresh()
                 {
                     // Everything else is fatal
                     //Crashes.TrackError(e);
-                    await Application.Current.MainPage.DisplayAlert("Login Failure", "There seems to have been a problem logging you in. Please try again. " + e.Message, "OK");
+                    await Application.Current.MainPage.DisplayAlert("Login Failure",
+                        "There seems to have been a problem logging you in. Please try again. " + e.Message, "OK");
+                }
+                finally
+                {
+                    isRunning = false;
+                    ButtonText = "Sign up / Log in";
+                    RaisePropertyChanged(nameof(ButtonText));
```

Comment:
I think you also need to raise property changed for isRunning and LoginButtonEnabled

Reply:
done